js/lib/ajax.js

```diff
@@ -1,261 +1,266 @@
 /*globals elgg, $*/
-elgg.provide('elgg.ajax');
-
-/**
- * @author Evan Winslow
- * Provides a bunch of useful shortcut functions for making ajax calls
- */
-
-/**
- * Wrapper function for jQuery.ajax which ensures that the url being called
- * is relative to the elgg site root.
- *
- * You would most likely use elgg.get or elgg.post, rather than this function
- *
- * @param {string} url Optionally specify the url as the first argument
- * @param {Object} options Optional. {@link jQuery#ajax}
- * @return {jqXHR}
- */
-elgg.ajax = function(url, options) {
-	options = elgg.ajax.handleOptions(url, options);
-
-	options.url = elgg.normalize_url(options.url);
-	return $.ajax(options);
-};
-/**
- * @const
- */
-elgg.ajax.SUCCESS = 0;
-
-/**
- * @const
- */
-elgg.ajax.ERROR = -1;
-
-/**
- * Handle optional arguments and return the resulting options object
- *
- * @param url
- * @param options
- * @return {Object}
- * @private
- */
-elgg.ajax.handleOptions = function(url, options) {
-	var data_only = true,
-		data,
-		member;
-
-	//elgg.ajax('example/file.php', {...});
-	if (elgg.isString(url)) {
-		options = options || {};
-
-	//elgg.ajax({...});
-	} else {
-		options = url || {};
-		url = options.url;
-	}
-
-	//elgg.ajax('example/file.php', function() {...});
-	if (elgg.isFunction(options)) {
-		data_only = false;
-		options = {success: options};
-	}
-
-	//elgg.ajax('example/file.php', {data:{...}});
-	if (options.data) {
-		data_only = false;
-	} else {
-		for (member in options) {
-			//elgg.ajax('example/file.php', {callback:function(){...}});
-			if (elgg.isFunction(options[member])) {
-				data_only = false;
+define(function(require) {
```

Comment:
Yea, it should, but you have an Elgg/ajax module already so I'm sorry of unsure where to put it.